R/hoist.R

```diff
@@ -114,27 +114,27 @@ hoist <- function(.data,
   )
 
   # Place new columns before old column
-  out <- append_df(.data, cols, after = match(.col, names(.data)) - 1L)
-
-  if (!.remove) {
-    return(out)
-  }
-
-  x <- map(x, function(x) {
-    # rev() is sneaky hack assuming that most people will remove in
-    # numeric order, so this should avoid most order problems. A full
-    # resolution will be considerably more work.
-    for (plucker in rev(pluckers)) {
-      x <- strike(x, plucker)
+  out <- df_append(.data, cols, after = match(.col, names(.data)) - 1L)
+
+  if (.remove) {
+    x <- map(x, function(x) {
+      # rev() is sneaky hack assuming that most people will remove in
+      # numeric order, so this should avoid most order problems. A full
+      # resolution will be considerably more work.
+      for (plucker in rev(pluckers)) {
+        x <- strike(x, plucker)
+      }
+      x
+    })
+
+    if (every(x, is_empty)) {
+      x <- NULL
     }
-    x
-  })
-  if (every(x, is_empty)) {
-    x <- NULL
+
+    out[[.col]] <- x
   }
-  out[[.col]] <- x
 
-  out
+  reconstruct_tibble(.data, out)
```

Comment:
This is new for `hoist()` but allows us to correctly restore grouped data frames. It is also now necessary because we use `df_append()` which will always return a bare data frame which likely needs to be restored to a tibble.